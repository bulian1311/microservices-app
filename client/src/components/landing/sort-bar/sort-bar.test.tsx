import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";
import useStore from "../../../hooks/use-store.hook";

import SortBar from "./sort-bar.component";

jest.mock("../../../api", () => {
  return {
    fetchProducts: jest.fn().mockResolvedValue({
      count: 1,
      products: [
        {
          id: 111,
          title: "Test product",
          description: "Test description",
          category: "test category",
          price: 123,
          url: "www",
          images: '["http://www."]',
          specifications: '{"111": "qqq"}',
          producer: "{}",
          tags: "test tag",
        },
      ],
    }),
  };
});

let storeHook = renderHook(() => useStore());

let documentBody: RenderResult;

beforeEach(() => {
  documentBody = render(<SortBar />);
  storeHook = renderHook(() => useStore());
});

test("Sort component, matches snapshot.", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
  expect(documentBody.getByText("сортировать")).toBeInTheDocument();
});

test("Sort component rerender, and show product count for store.", async () => {
  await storeHook.result.current.productStore.fetchProducts(false);

  expect(documentBody.getByText("Всего аттракционов: 1")).toBeInTheDocument();
});
