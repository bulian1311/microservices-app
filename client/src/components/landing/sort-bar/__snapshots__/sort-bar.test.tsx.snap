// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component, matches snapshot. 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz bCMlzV"
    >
      <span
        class="sc-dlfnbm eWQShC"
      >
        Всего
         аттракционов:
         
        0
      </span>
      <span
        class="sc-dlfnbm eWQShC"
      >
        <i
          class="sc-hKgILt iVaAuo"
        >
          <svg
            class="sc-bdfBwQ hDpvxr"
            viewBox="0 0 24 16"
            width="1.5"
          >
            <path
              d="M9.33333 16H14.6667V13.3333H9.33333V16ZM0 0V2.66667H24V0H0ZM4 9.33333H20V6.66667H4V9.33333Z"
              fill="#555555"
            />
          </svg>
        </i>
        сортировать
      </span>
    </div>
  </div>
</body>
`;
